omegaconf/_utils.py

```diff
@@ -454,6 +439,16 @@ def is_primitive_container(obj: Any) -> bool:
     return is_primitive_list(obj) or is_primitive_dict(obj)
 
 
+def get_union_types(ref_type: Optional[Any]) -> List[Any]:
+    args = getattr(ref_type, "__args__", None)
+    element_types: List[Any]
+    if ref_type is not Union and args is not None and args[0] is not Any:
+        element_types = list(args)
+    assert element_types is not None
```

Comment:
> Not sure what the if ref_type is not Union test is supposed to catch: if I remove it, all tests still pass

It tested Unions without types(which don't make sense, like List or Dict alone now that I think about it). 

I think this should be better to return Any in case it's not a union like get_element_list. Correct me if I'm wrong.

PS: Thanks for your feedback!

Reply:
> > Not sure what the if ref_type is not Union test is supposed to catch: if I remove it, all tests still pass
> 
> It tested Unions without types(which don't make sense, like List or Dict alone now that I think about it).
> 
> I think this should be better to return Any in case it's not a union like get_element_list. Correct me if I'm wrong.

Ah ok, personally I think it'd be best to raise an exception if this function is called with anything that's not a `Union[something]`, because I can't think of any use case where this would be intended.
Also I think that `something` should have at least two elements.

Reply:
What is a union without a type?
like this?

```
@dataclass
class X:
 a: Union
```

This seems like a legitimate Python (at least it runs and mypy --strict does not complain).
not sure what it's supposed to mean though. Unless someone can tell me what this is for, I think we should raise an error for a Union without a list of types.

Reply:
> I think we should raise an error for a Union without a list of types.

Agreed -- just tested and currently it's accepted.